File: ruby/leap/leap.rb
class Year
  class << self
    def leap?(year)
      @year = year
      (divisible_by(4) && !divisible_by(100)) || divisible_by(400)
    end

    private

    def divisible_by(number)
      (@year % number).zero?
    end
  end
end

module BookKeeping
  VERSION = 3
end

File: javascript/leap/leap.js
function Year(input) {
    this.year = input;
};

Year.prototype.divisible_by = function(number) {
    return (this.year % number) == 0
};

Year.prototype.isLeap = function() {
    return (this.divisible_by(4) && !this.divisible_by(100)) || this.divisible_by(400);
};

module.exports = Year;
